graphics/src/ColladaExporter_TEST.cc

```diff
@@ -315,6 +315,151 @@ TEST_F(ColladaExporter, ExportMeshWithSubmeshes)
+
+  auto lib_lights = collada->FirstChildElement("library_lights");
+  EXPECT_TRUE(lib_lights);
+
+  int light_count = 0;
+  auto light_ele = lib_lights->FirstChildElement("light");
+  for (; light_ele != nullptr; light_ele = light_ele->NextSiblingElement())
+  {
+    const char* light_name_cstr = light_ele->Attribute("name");
+    ASSERT_TRUE(light_name_cstr);
+    std::string light_name = light_name_cstr;
+    if (light_name == "sun")
+    {
+      auto technique = light_ele->FirstChildElement("technique_common");
+      EXPECT_TRUE(technique);
+      auto directional = technique->FirstChildElement("directional");
+      EXPECT_TRUE(directional);
+      auto color = directional->FirstChildElement("color");
+      EXPECT_TRUE(color);
+    }
+    else if (light_name == "lamp")
+    {
+      auto technique = light_ele->FirstChildElement("technique_common");
+      EXPECT_TRUE(technique);
+      auto point = technique->FirstChildElement("point");
+      EXPECT_TRUE(point);
+      auto color = point->FirstChildElement("color");
+      EXPECT_TRUE(color);
+      auto catt = point->FirstChildElement("constant_attenuation");
+      EXPECT_TRUE(catt);
+      auto latt = point->FirstChildElement("linear_attenuation");
+      EXPECT_TRUE(latt);
+      auto qatt = point->FirstChildElement("quadratic_attenuation");
+      EXPECT_TRUE(qatt);
+    }
+    else if (light_name == "torch")
+    {
+      auto technique = light_ele->FirstChildElement("technique_common");
+      EXPECT_TRUE(technique);
+      auto spot = technique->FirstChildElement("spot");
+      EXPECT_TRUE(spot);
+      auto color = spot->FirstChildElement("color");
+      EXPECT_TRUE(color);
+      auto catt = spot->FirstChildElement("constant_attenuation");
+      EXPECT_TRUE(catt);
+      auto latt = spot->FirstChildElement("linear_attenuation");
+      EXPECT_TRUE(latt);
+      auto qatt = spot->FirstChildElement("quadratic_attenuation");
+      EXPECT_TRUE(qatt);
+      auto falloff_angle = spot->FirstChildElement("falloff_angle");
+      EXPECT_TRUE(falloff_angle);
+      auto falloff_expo = spot->FirstChildElement("falloff_exponent");
+      EXPECT_TRUE(falloff_expo);
+    }
+    else
+      ASSERT_TRUE(0);  // Invalid light name given
+
+    ++light_count;
+  }
+  EXPECT_EQ(light_count, 3);
+
+  // instantiation
+  auto lib_visual_scenes = collada->FirstChildElement("library_visual_scenes");
+  EXPECT_TRUE(lib_visual_scenes);
+  auto scene = lib_visual_scenes->FirstChildElement("visual_scene");
+  EXPECT_TRUE(scene);
+
+  int node_with_light_count = 0;
+  auto node_ele = scene->FirstChildElement("node");
+  for (; node_ele != nullptr; node_ele = node_ele->NextSiblingElement())
+  {
+    const char* node_name_cstr = node_ele->Attribute("name");
+    ASSERT_TRUE(node_name_cstr);
+    std::string node_name = node_name_cstr;
+    if (node_name == "sun" || node_name == "lamp" || node_name == "torch")
+    {
+      auto inst_light = node_ele->FirstChildElement("instance_light");
```

Comment:
do you need these variables ?

Reply:
it's good to just check for the tags no?

Reply:
I'm fine with the checks, I was just suggesting something like:

```
      EXPECT_TRUE(node_ele->FirstChildElement("translate"));
      EXPECT_TRUE(node_ele->FirstChildElement("rotate"));
```

Reply:
remove 1b7602b